File: Ad-Hoc/2463/2463.cs
using System;
using System.Linq;

class URI 
{
    static void Main()
    {
        const int n_salas = int.Parse(Console.ReadLine());
        const int[] salas = Console.ReadLine().Split(' ').Select(int.Parse).ToArray();

        int local_max = salas[0];
        int global_max = salas[0];

        for(int i = 1; i < n_salas; i++){
            int curr = salas[i];
            local_max = Math.Max(local_max + curr, curr);
            if(local_max > global_max){
                global_max = local_max;
            }
        }

        Console.WriteLine(global_max);
    }
}

File: Ad-Hoc/2463/2463.cpp
#include <iostream>
#include <vector>
#include <cmath>

using namespace std;

int main(){

    int n;
    cin >> n;

    vector<int> arr(n);
    for(size_t i = 0; i < n; i++){
        cin >> arr[i];
    }

    int local_max = arr[0];
    int global_max = arr[0];

    for(size_t i = 1; i < n; i++){
        local_max = max(arr[i], local_max + arr[i]);
        if(local_max > global_max){
            global_max = local_max;
        }
    }

     cout << global_max << endl;
     return 0;
}